File: 02_Sort_Application/Level4/2075.cpp
// Copyright@2023 Jihoon Lucas Kim <jihoon.lucas.kim@gmail.com>
// N번째 큰 수
// https://www.acmicpc.net/problem/2075
// 힌트
// 1. 모든 수에 대해 내림차순 정렬 후 N-1 인자 출력
// * cpp의 경우 cin.tie(NULL); ios_base::sync_with_stdio(false);를 작성해주지 않으면 시간 초과발생
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    cin.tie(NULL);
    ios_base::sync_with_stdio(false);
    vector<int> v;
    int N;
    cin >> N;
    v = vector<int>(N * N);
    for (int i = 0; i < N * N; i++)
    {
        cin >> v[i];
    }
    sort(v.begin(), v.end(), greater<int>());
    cout << v[N - 1] << "\n";

    return 0;
}

File: 02_Sort_Application/Level4/2075.java
// Copyright@2023 Jihoon Lucas Kim <jihoon.lucas.kim@gmail.com>
// N번째 큰 수
// https://www.acmicpc.net/problem/2075
// 힌트
// 1. 모든 수에 대해 정렬 후 끝에서 N번째 인자 출력
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int N = Integer.parseInt(br.readLine());
		int[] v = new int[N * N];
		
		StringTokenizer st;
		for (int i = 0; i < N ; i++) {
			st = new StringTokenizer(br.readLine());
			for (int j = 0 ; j < N; j++) {
				v[i*N + j] = Integer.parseInt(st.nextToken());
			}
		}
		
		Arrays.sort(v);
		
		System.out.println(v[N * N - N]);
	}
}
